File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/media/files/stored/StoredFileSystemFileProducer.java
package com.lasthopesoftware.bluewater.client.library.items.media.files.stored;

import android.support.annotation.NonNull;

import com.lasthopesoftware.bluewater.client.library.items.media.files.stored.repository.StoredFile;

import java.io.File;

public final class StoredFileSystemFileProducer implements IStoredFileSystemFileProducer {

	@Override
	public File getFile(@NonNull StoredFile storedFile) {
		return new File(storedFile.getPath());
	}

}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/items/files/StoredFileSystemFileProducer.kt
package com.lasthopesoftware.bluewater.client.stored.library.items.files

import com.lasthopesoftware.bluewater.client.stored.library.items.files.repository.StoredFile
import java.io.File

class StoredFileSystemFileProducer : IStoredFileSystemFileProducer {
    override fun getFile(storedFile: StoredFile): File? {
        return storedFile.path?.let(::File)
    }
}
